gcloud/error_reporting/client.py

```diff
@@ -0,0 +1,142 @@
+
+    :raises: :class:`ValueError` if the project is neither passed in nor
+             set in the environment.
+    """
+
+    def __init__(self, project=None,
+                 credentials=None,
+                 http=None,
+                 service=None,
+                 version=None):
+        self.logging_client = gcloud.logging.client.Client(
+            project, credentials, http)
+        self.service = service
+        self.version = version
+
+    DEFAULT_SERVICE = 'python'
+
+    def _get_default_service(self):
+        """Returns the service to use on method calls that don't specify an
+        override.
+
+          :rtype: string
+          :returns: The default service for error reporting calls
+        """
+        if self.service:
+            return self.service
+        else:
+            return self.DEFAULT_SERVICE
+
+    def _get_default_version(self):
+        """Returns the service to use on method calls that don't specify an
+        override.
+
+        :rtype: string
+        :returns: The default version for error reporting calls.
+        """
+        if self.version:
+            return self.version
+
+    def report(self, message="", service=None, version=None):
+        """ Reports the details of the latest exceptions to Stackdriver Error
+            Reporting.
+
+         https://cloud.google.com/error-reporting/docs/formatting-error-messages
+
+           :type message: str
+           :param message: An optional message to include with the exception
+                           detail
+
+           :type service: str
+           :param service: An identifier of the service, such as the name of
+                           the executable, job, or Google App Engine service
+                           name. This field is expected to have a low number
+                           of values that are relatively stable over time,
+                           as opposed to version, which can be changed
+                           whenever new code is deployed.
+
+           :type version: str
+           :param version: Represents the source code version that the
+                           developer provided, which could represent a
+                           version label or a Git SHA-1 hash, for example. If
+                           the developer did not provide a version, the value
+                           is set to default.
+
+
+           Example::
+
+                >>>     try:
+                >>>         raise NameError
+                >>>     except Exception:
+                >>>         client.report("Something went wrong!")
+        """
+        if not service:
+            service = self._get_default_service()
+        if not version:
+            version = self._get_default_version()
+        payload = {
+            'serviceContext': {
+                'service': service,
+            },
+            'message': '{0} : {1}'.format(message, traceback.format_exc())
```

Comment:
Message should not contain a combination of a user provided message and an exception.
See the 'Python' tab of https://cloud.google.com/error-reporting/docs/setup/compute-engine

it should instead just be `'message': '{0}'.format(traceback.format_exc())`